controllers/lighting_controller/lighting_controller.py

```diff
@@ -0,0 +1,270 @@
+"""
+The controller for altering arena lighting provided by a DirectionalLight and a Background
+Currently doesn't support:
+- Timed pre-match lighting changes
+"""
+import sys
+from typing import Dict, List, Tuple, Optional, NamedTuple
+from pathlib import Path
+from dataclasses import dataclass
+
+from controller import Node, Supervisor
+
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+
+sys.path.insert(1, str(REPO_ROOT / 'modules'))
+
+import controller_utils  # isort:skip
+import webots_utils  # isort:skip
+
+
+class ArenaLighting(NamedTuple):
+    lightDef: str
+    intensity: float = 3.5
+    colour: Tuple[float, float, float] = (1, 1, 1)
+
+
+class LightingEffect(NamedTuple):
+    start_time: float  # Negative start times are relative to the end of the match
+    # Negative values 0 - (-0.08) don't get included in the produced recordings
+    fade_time: Optional[float] = None
+    lighting: List[ArenaLighting] = [ArenaLighting('SUN')]
+    luminosity: float = 0.35
+    name: str = ""
+
+    def __repr__(self) -> str:
+        lights_info = [
+            f"({light.lightDef}, int={light.intensity}, col={light.colour})"
+            for light in self.lighting
+        ]
+        return (
+            f"<LightingEffect: \"{self.name}\", "
```

Comment:
Suggest using `!r` rather than manually adding quotes. (Note also that a `__repr__` probably deserves `!r` anyway).

Reply:
7e6e72380fc69df1ba61efaef574d095c474fc65